libs/barista-components/chart/src/heatfield/chart-heatfield.spec.ts

```diff
@@ -189,6 +197,27 @@ describe('DtChartHeatfield', () => {
       });
     });
 
+    describe('Closing behavior', () => {
+      let closeOverlayButton;
+      beforeEach(() => {
+        instance.isActive = true;
+        fixture.detectChanges();
+        closeOverlayButton = overlayContainerElement.querySelector(
+          '.dt-heatfield-close-trigger',
+        );
+      });
+
+      it('should include an x button', () => {
+        expect(closeOverlayButton).not.toBeNull();
+      });
+
+      it('clicking on x button should close overlay', () => {
+        closeOverlayButton.click();
+        fixture.detectChanges();
+        expect(marker.classList).not.toContain('dt-chart-heatfield-active');
+      });
+    });
```

Comment:
It would be awesome if you could add a testcase for the escape behavior

Reply:
Already thought about it after lukas's comment but let's fix the trap focus first :)